src/workflow/StreamlitUI.py

```diff
@@ -131,8 +133,18 @@ def upload_widget(
                                 shutil.copy(f, Path(files_dir, f.name))
                             else:
                                 symlink_target = Path(files_dir, f.name)
-                                if not symlink_target.exists():
-                                    os.symlink(f, symlink_target)
+                                if OS_PLATFORM == "win32":
+                                    # Detect if it's a file or directory
+                                    if f.is_file():
+                                        if not symlink_target.exists():
+                                            os.link(f, symlink_target)
+                                    else:
+                                        # Use mklink /J for directories
+                                        if not symlink_target.exists():
+                                            subprocess.call(["mklink", "/J", str(symlink_target), str(f)], shell=True)
+                                else:
+                                    if not symlink_target.exists():
+                                        os.symlink(f, symlink_target)
```

Comment:
@t0mdavid-m Is it possible to handle window's "symlink" using `os.link` for files and `mlink` for directories? I tested this on a windows machine, and it seemed to work and left the original files intact after deleting the linked/junction files.

Reply:
`os.link` creates a hardlink to the file. After giving it some thought, there are a couple points that speak against the usage of hard links:

- Hard links only work on the same file system (i.e. if the WebApp is on `C:\` and the file on `D:\` it is not possible to create a hardlink).
- If the user deletes the file from their drive, the storage is not released until the hardlink is also deleted.
- Some file systems (e.g. `FAT`) do not support hardlinks.

Thus, I would be in favour of storing the path instead.